gluten-core/src/main/java/io/glutenproject/substrait/rel/ProjectRelNode.java

```diff
@@ -31,27 +31,36 @@ public class ProjectRelNode implements RelNode, Serializable {
   private final ArrayList<ExpressionNode> expressionNodes =
       new ArrayList<>();
   private final AdvancedExtensionNode extensionNode;
+  private final int emitStartIndex;
 
   ProjectRelNode(RelNode input,
-                 ArrayList<ExpressionNode> expressionNodes) {
+                 ArrayList<ExpressionNode> expressionNodes,
+                 int emitStartIndex) {
     this.input = input;
     this.expressionNodes.addAll(expressionNodes);
     this.extensionNode = null;
+    this.emitStartIndex = emitStartIndex;
   }
 
   ProjectRelNode(RelNode input,
                  ArrayList<ExpressionNode> expressionNodes,
-                 AdvancedExtensionNode extensionNode) {
+                 AdvancedExtensionNode extensionNode,
+                 int emitStartIndex) {
     this.input = input;
     this.expressionNodes.addAll(expressionNodes);
     this.extensionNode = extensionNode;
+    this.emitStartIndex = emitStartIndex;
   }
 
   @Override
   public Rel toProtobuf() {
     RelCommon.Builder relCommonBuilder = RelCommon.newBuilder();
-    relCommonBuilder.setDirect(RelCommon.Direct.newBuilder());
-
+    // relCommonBuilder.setDirect(RelCommon.Direct.newBuilder());
+    RelCommon.Emit.Builder emitBuilder = RelCommon.Emit.newBuilder();
+    for (int i = 0; i < expressionNodes.size(); i++) {
```

Comment:
How about still setting direct if `emitStartIndex` is invalid or is not set?

Reply:
Thanks, just fixed it.